### PR TITLE
VFS: Fix mounting non-existant paths

### PR DESCRIPTION
**Pull request by elad335**
resolve_path is just a simplification of path compariosns I added in the past for savestates. It is not  
 required to succeed, nor should VFS care whether the file path exists or not.

### CODE REVIEW

**Comment by Megamouse** [comment]
Shouldn't this be handled in resolve_path?
The current way is super unintuitive and undocumented.
No one knows this behaviour except you and me

**Comment by elad335** [comment]
There was a pr to use std::filesystem::weakly_canonical which claims to not have this issue, but it got closed due to a suspection of a bug.

**Comment by Megamouse** [comment]
At least add a comment to the callback or whatever

File: rpcs3/main_application.cpp [hunk 1]
```diff
@@ -327,7 +328,15 @@ EmuCallbacks main_application::CreateCallbacks()
 
 	callbacks.resolve_path = [](std::string_view sv)
 	{
-		return QFileInfo(QString::fromUtf8(sv.data(), static_cast<int>(sv.size()))).canonicalFilePath().toStdString();
+		const std::string result = QFileInfo(QString::fromUtf8(sv.data(), static_cast<int>(sv.size()))).canonicalFilePath().toStdString();
+
+		if (result.empty())
+		{
+			// May fail if path does not exist
+			std::clog << fmt::format("main_application: resolve_path(): Failed to resolve path \'%s\'\n", sv);
```
Comment (Megamouse) [inline rpcs3/main_application.cpp:336]: No idea what this would help

File: rpcs3/main_application.cpp [hunk 2]
```diff
@@ -37,6 +37,7 @@
 #include <QFileInfo> // This shouldn't be outside rpcs3qt...
 #include <QImageReader> // This shouldn't be outside rpcs3qt...
 #include <thread>
+#include <iostream>
```
Comment (Megamouse) [inline rpcs3/main_application.cpp:40]: !

**Pull request merged**